### ISSUE TITLE
Using flexible attribute in path definition

### ISSUE THREAD

**Issue by PierreRust** [beetbox/beets#816]
This is a follow-up from a [discussion on the mailing-list](https://groups.google.com/forum/#!topic/beets-users/cuZ-hvk2prU).

When using flexible attributes in path definition, missing fields are not expanded, meaning that they appear in the result as they were defined in the path. For example : 

```
beet ls -f '$test_attr' album:Madame
```

Could return someting like : 

```
flexattr_value
$test_attr
$test_attr
flexattr_value
```

The current behavior has the following advantages : 
- It provides a way to differentiate between a missing and empty field.
- It makes debugging path templates easier : you can easily see if you misspelled a field rather than just having it disappear.
- Symmetry with Python's built-in templating.

One the other hand, this can be quite annoying as you generally do not want the `$'field_name'` string to appear in your path and missing fields are very common when using flexible attributes.

They seems to be several option to make this simpler : 
- create a new construct for path definition %ifdef{field,true,false}
- output an empty string for the missing field, one can always use the already existing %if construct to specify a default value when needed.

**Comment by GuilhermeHideki** [comment]
I did a "quick fix" with this: 
**TL;DR** this won't work if your value starts with '$'
```py
class IfDefPlugin(BeetsPlugin):
    def __init__(self):
        super(IfDefPlugin, self).__init__()

        self.template_funcs['ifdef'] = self.ifdef_template_func

    def ifdef_template_func(self, text):
        return 0 if text is None or not text or text.startswith('$') else 1
```
since returns 0 or 1, usage should be like ``%if{%ifdef{$series},$series/}$album``, for example.
this would be better the attribute could return none instead of "${attribute}"

**Comment by sampsyo** [comment]
Cool! FWIW, `text is None or not text` is redundant; it's exactly the same as `not text`.

**Comment by awesomer** [comment]
FTR, I'm happy with the %ifdef solution, though I think I agree with sampsyo on IRC :

> < sampsyo> I guess I'd advocate for something like %ifdef{field} instead of %ifdef{$field}
> < sampsyo> Which would skip the default behavior of expanding $field to its value and directly check whether it's defined.
